hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/RSProcedureDispatcher.java

```diff
@@ -330,6 +322,73 @@ private boolean scheduleForRetry(IOException e) {
       return true;
     }
 
+    private boolean unableToConnectToServerInFirstAttempt(IOException e) {
```

Comment:
I mean we could extract a method for testing exception only, and then we test numberOfAttemptsSoFar outside the method...